### PR TITLE
Refactor LoginForm with RHF validation

### PR DESCRIPTION
**Pull request by thienchi2109**
## Summary
- Refactor LoginForm to use react-hook-form, zod, and shadcn Form primitives
- Add localized login validation/server-error messages
- Cover empty validation, server error mapping, autocomplete, and successful submit behavior

## Verification
- node scripts/npm-run.js run verify:no-explicit-any
- node scripts/npm-run.js run typecheck
- node scripts/npm-run.js exec vitest run src/app/_components/__tests__/LoginForm.test.tsx src/app/__tests__/page.authenticated-redirect.test.tsx
- node scripts/npm-run.js npx react-doctor@latest . --verbose -y --project nextn --offline --diff main

Fixes #368

---
## Summary by cubic
Refactors the login form to use `react-hook-form` with `zod` and shadcn `Form`, adding localized field/server error messages and better a11y. Fixes #368 by enforcing empty-field checks, mapping backend error codes, and expanding tests.

- **Refactors**
  - Replace local state with `react-hook-form` + `@hookform/resolvers/zod`; field-level validation with i18n messages.
  - Map `tenant_inactive`, `rpc_error`, and `invalid_credentials` to localized messages; show in `role="alert"` with `aria-live="polite"`.
  - Keep password manager hints via `autocomplete="username"` and `"current-password"`.
  - Switch to shadcn `Form`, `Input`, `Button`; remove mount-only animation and add `suppressHydrationWarning` on footer year.
  - Add tests for empty validation, server error mapping (incl. unexpected failures), autocomplete, and successful submit; add new EN/VI i18n keys.

<sup>Written for commit 523822cf37c6642b320f65ea624700640bcdea80. Summary will update on new commits.</sup>

## Release Notes

* **Improvements**
  * Login form now displays clearer, inline validation error messages in multiple languages.
  * Authentication failures provide more informative and helpful error feedback to users.
  * Enhanced form validation and error handling for a better login experience.
  * Expanded multi-language support for all login-related messages and error states.

* **Tests**
  * Added comprehensive automated tests for login form functionality and error scenarios.

### CODE REVIEW

File: src/app/_components/__tests__/LoginForm.test.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  signIn: (...args: unknown[]) => mocks.signIn(...args),
+}))
+
+vi.mock("@/components/icons", () => ({
+  Logo: () => "Logo",
+}))
+
+vi.mock("../LoginIllustrationPanel", () => ({
+  LoginIllustrationPanel: () => null,
+}))
```
[the remark this answers is not in the document]
Comment (thienchi2109) [reply src/app/_components/__tests__/LoginForm.test.tsx]: Fixed in 523822c: changed the LoginIllustrationPanel mock to use the `@/app/_components/LoginIllustrationPanel` alias.

File: src/app/_components/__tests__/LoginForm.test.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+function renderLoginForm() {
+  return render(
+    <LanguageProvider>
+      <LoginForm />
+    </LanguageProvider>,
+  )
+}
```
[the remark this answers is not in the document]
Comment (thienchi2109) [reply src/app/_components/__tests__/LoginForm.test.tsx]: Fixed in 523822c: `renderLoginForm` now declares `ReturnType<typeof render>`.

File: src/app/_components/LoginForm.tsx [hunk 3]
```diff
@@ -1,50 +1,116 @@
...
 import { useLanguage } from "@/contexts/language-context"
 import { LoginIllustrationPanel } from "./LoginIllustrationPanel"
 
+function getLoginMessage(
+  t: (key: string) => string | undefined,
+  key: string,
+  fallback: string,
+) {
+  return t(key) || fallback
+}
```
[the remark this answers is not in the document]
Comment (thienchi2109) [reply src/app/_components/LoginForm.tsx:29]: Fixed in 523822c: added explicit return types for `getLoginMessage`, `createLoginFormSchema`, `getCredentialErrorMessage`, `LoginForm`, and `handleLogin`.

File: src/app/_components/LoginForm.tsx [hunk 4]
```diff
@@ -1,50 +1,116 @@
...
+      // Success: NextAuth session established; no legacy bridge writes.
       window.location.href = "/dashboard"
     } catch (err) {
       console.error("Unexpected login error", { error: err })
-      setError(t("login.error") || "Tên đăng nhập hoặc mật khẩu không đúng")
-      setIsLoading(false)
+      form.setError("root", {
+        type: "server",
+        message: getCredentialErrorMessage(t, undefined),
+      })
```
[the remark this answers is not in the document]
Comment (thienchi2109) [reply src/app/_components/LoginForm.tsx:111]: Fixed in 523822c: unexpected `signIn` exceptions now map to the `rpc_error`/system message instead of the invalid-credentials fallback. Added a focused regression test for a rejected `signIn` call.

File: src/app/_components/LoginForm.tsx [hunk 5]
```diff
@@ -1,50 +1,116 @@
...
+
+      // Success: NextAuth session established; no legacy bridge writes.
       window.location.href = "/dashboard"
     } catch (err) {
       console.error("Unexpected login error", { error: err })
-      setError(t("login.error") || "Tên đăng nhập hoặc mật khẩu không đúng")
-      setIsLoading(false)
+      form.setError("root", {
+        type: "server",
+        message: getCredentialErrorMessage(t, undefined),
```
[the remark this answers is not in the document]
Comment (thienchi2109) [reply src/app/_components/LoginForm.tsx]: Fixed in 523822c: the catch path now uses `getCredentialErrorMessage(t, "rpc_error")`, with a regression test covering an unexpected rejected `signIn` call.
